### PR TITLE
[MM-26471] Team Profile Setup step

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Team Profile Setup step.
Fully functional.
Includes some changes to the common components to facilitate features we needed for this work.

Couple notes:
- The new team icon doesn't seem to refresh until an F5 is pressed. Will look into why, but I'm guessing it's a bug with that component.
- Missing the Team URL component as part of a discussion with @matthewbirtch and @jwilander.
- There is a bug on resizing the window where the card doesn't resize. It's fixed in #6077.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-26471

### CODE REVIEW

**Comment by matthewbirtch** [comment]
@devinbinnie before I test further, I'm wondering if something is broken. The input fields have the wrong padding on the left and I know this is more about the Team profile, but the user profile steps looks broken too.

image
user profile image is stacked below the display name field and the name field is really wide.

image

Notice all the extra space to the left of the text inside the input fields.

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    }
+
+    onSkip = () => {
+        this.props.onSkip(this.props.id);
+    }
+
+    onFinish = () => {
+        const team = Object.assign({}, this.props.team, {display_name: this.state.teamName});
+
+        this.props.actions.patchTeam(team);
```
Comment (reflog) [inline components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:96]: no need for await here?

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+        this.setState({teamURL: event.target.value, teamURLError});
+    }
+
+    isFinishDisabled = () => {
+        return Boolean(!this.state.teamName || this.state.teamNameError || !this.state.teamURL || this.state.teamURLError || this.state.profilePictureError);
+    }
+
+    getSanitizedSiteURL = () => {
+        const removeProtocol = this.props.siteURL.replace(/(^\w+:|^)\/\//, '');
+        if (removeProtocol[removeProtocol.length - 1] === '/') {
```
Comment (reflog) [inline components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:82]: nit: removeProtocol.endsWith('/') is cleaner

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx (hunk 1 above)
Comment (devinbinnie) [reply components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:96]: Nope, we can fire and forget. There's no component state that depends on the outcome.

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx (hunk 2 above)
Comment (devinbinnie) [reply components/next_steps_view/steps/team_profile_step/team_profile_step.tsx]: Yes, yes it is :P

**Comment by devinbinnie** [comment]
> @devinbinnie before I test further, I'm wondering if something is broken. The input fields have the wrong padding on the left and I know this is more about the Team profile, but the user profile steps looks broken too.
> 
> image
> user profile image is stacked below the display name field and the name field is really wide.
> 
> image
> 
> Notice all the extra space to the left of the text inside the input fields.

@matthewbirtch Sorry about that, there's some fixes needed from the other branch that fix the styling. I've brought those fixes over to this branch so you can review.

**Comment by stevemudie** [comment]
- I'm unable to type in anything in step 1 without it showing an error
- It will accept continuing with it blank though
- The 'Invite members' cad (3) shows an error when any text is typed

**Comment by devinbinnie** [comment]
> * I'm unable to type in anything in step 1 without it showing an error
> * It will accept continuing with it blank though
> * The 'Invite members' cad (3) shows an error when any text is typed

Both the Complete Profile step and Invite Members steps aren't working in this PR, Complete Profile is working here: https://github.com/mattermost/mattermost-webapp/pull/6077
Invite Members is being worked on currently and will be in another PR.

**Comment by matthewbirtch** [comment]
@devinbinnie is it possible to update the team name in the sidebar once you've saved? It doesn't appear to be updating
Screen Shot 2020-08-10 at 10 03 41 AM

Related to the other note, the styling on the 'name your team' step looks fixed, but not on the other 2 steps.

**Comment by devinbinnie** [comment]
> @devinbinnie is it possible to update the team name in the sidebar once you've saved? It doesn't appear to be updating

Should be working, can you test it once the update is done?

> Related to the other note, the styling on the 'name your team' step looks fixed, but not on the other 2 steps.

Fixed!

> 
> Also, the jumpiness inside the accordions is back. When opening and closing each accordion, the content within seems to jump around a bit and doesn't make it feel smooth - not sure if that ever got fixed actually or if you were planning to do it in another PR.

That should be in another PR. I have a ticket open to do it.

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    onSkip = () => {
+        this.props.onSkip(this.props.id);
+    }
+
+    onFinish = () => {
+        const team = Object.assign({}, this.props.team, {display_name: this.state.teamName});
+
+        this.props.actions.patchTeam(team);
+
+        // TODO: patch url?
```
Comment (jwilander) [inline components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:98]: Comment still needed?

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+                            <Input
+                            name='teamURL'
+                            type='text'
+                            value={this.state.teamURL}
+                            onChange={this.handleURLInputChange}
+                            placeholder={Utils.localizeMessage('next_steps_view.team_profile_step.teamURL', 'Team URL')}
+                            error={this.state.teamURLError}
+                            info={Utils.localizeMessage('next_steps_view.team_profile_step.youCanCustomize', 'You can customize your team url')}
+                            textPrefix={this.getSanitizedSiteURL()}
+                        /> */}
```
Comment (jwilander) [inline components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:179]: Should be removed right?

**Comment by matthewbirtch** [comment]
> > @devinbinnie is it possible to update the team name in the sidebar once you've saved? It doesn't appear to be updating
> 
> Should be working, can you test it once the update is done?

Still doesn't appear to be working for me - even after saving and then refreshing

**Comment by devinbinnie** [comment]
> > > @devinbinnie is it possible to update the team name in the sidebar once you've saved? It doesn't appear to be updating
> > 
> > 
> > Should be working, can you test it once the update is done?
> 
> Still doesn't appear to be working for me - even after saving and then refreshing

Are you a team admin on the team you're changing the name for?
I just made a new user and a new team for that user and it worked for me.

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx (hunk 3 above)
Comment (devinbinnie) [reply components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:98]: Nahh I'll remove it.

File: components/next_steps_view/steps/team_profile_step/team_profile_step.tsx (hunk 4 above)
Comment (devinbinnie) [reply components/next_steps_view/steps/team_profile_step/team_profile_step.tsx:179]: I guess for now yeah I'll remove it, wasn't sure if we were gonna bring it back once we have a plan for actually doing this (if it's even possible).

**Comment by matthewbirtch** [comment]
> Are you a team admin on the team you're changing the name for?
> I just made a new user and a new team for that user and it worked for me.

My apologies @devinbinnie that appeared to be the issue. Instead of creating a new team, I think I chose an existing team and then tried to change it and it didn't work.

But I just tried creating a new team and all is well. I assume we wouldn't run in to the issue above for a new admin, so we should be ok

**Comment by devinbinnie** [comment]
@stevemudie Do you have a moment to review?

**Pull request merged**